internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,445 @@
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/cpstructs"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/form"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+)
+
+const (
+	invalidDirectoryErrorMsg       = `plugin directory "%s" does not exist`
+	unexpectedInstallationErrorMsg = "unexpected installation type: %s"
+)
+
+func (c *pluginCommand) newInstallCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install <plugin>",
+		Short: "Install a Connect plugin.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.install,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Install the latest release of the Datagen connector onto your local Confluent Platform environment.",
+				Code: "confluent connect plugin install confluentinc/kafka-connect-datagen:latest",
+			},
+			examples.Example{
+				Text: "Install the latest release of the Datagen connector in a user specified directory and update one worker-config file.",
+				Code: "confluent connect plugin install confluentinc/kafka-connect-datagen:latest --plugin-directory $CONFLUENT_HOME/plugins --worker-configs $CONFLUENT_HOME/etc/kafka/connect-distributed.properties",
+			},
+		),
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireOnPremLogin},
+	}
+
+	cmd.Flags().String("plugin-directory", "", "The plugin installation directory. If not specified, a default will be selected based on your Confluent Platform installation.")
+	cmd.Flags().StringSlice("worker-configs", []string{}, "A comma-separated list of paths to one or more Kafka Connect worker configuration files. Each worker file will be updated to load plugins from the plugin directory in addition to any preexisting directories.")
+	cmd.Flags().String("confluent-platform", "", "The path to a Confluent Platform archive installation. If not specified, this command will search for Confluent Platform installations in common locations.")
+	pcmd.AddDryRunFlag(cmd)
+	cmd.Flags().Bool("force", false, "Proceed without user input.")
+
+	cobra.CheckErr(cmd.MarkFlagDirname("plugin-directory"))
+
+	return cmd
+}
+
+func (c *pluginCommand) install(cmd *cobra.Command, args []string) error {
+	dryRun, err := cmd.Flags().GetBool("dry-run")
+	if err != nil {
+		return err
+	}
+
+	if err := c.InitializeHubClient(); err != nil {
+		return err
+	}
+
+	pluginManifest, err := c.getManifest(args[0])
+	if err != nil {
+		return err
+	}
+
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+
+	if cmd.Flags().Changed("plugin-directory") && cmd.Flags().Changed("worker-configs") && cmd.Flags().Changed("confluent-platform") {
+		return errors.New("at most two of `--plugin-directory`, `--worker-configs`, and `--confluent-platform` may be set")
```

Comment:
This seems like a weird requirement. So any of the three combinations of two flags can be passed here?

Reply:
The `--confluent-platform` flag gets ignored completely if both of the other flags are specified, which is why I error out if all three are there. The platform installation is only used as a helper to suggest values to the user for the other two if the user didn't provide them.

The user can provide zero, one, or two of the flags.